### DISCUSSION TITLE
Does CPU and GPU version of ArcFace produce the same vectors? Can I cope vectors from GPU version to calculate distance using CPU only version?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [exadel-inc/CompreFace#875] [Q&A] [answered]
Does CPU and GPU version of ArcFace produce the same vectors? Can I cope vectors from GPU version to calculate distance using CPU only version?

**Comment by pospielov** [comment, marked as answer]
Theoretically yes. I mean, the models are the same in GPU and CPU versions. And when I checked, it looks like they really work the same. 
Still, I've heard that the precision of CPU and GPU may be different (like numbers after 0.0000000001, for example).
